//
// PROJECT:         Aspia
// FILE:            system_info/category_dmi_onboard_device.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message DmiOnBoardDevice
{
    enum Type
    {
        TYPE_UNKNOWN         = 0;
        TYPE_OTHER           = 1;
        TYPE_VIDEO           = 2;
        TYPE_SCSI_CONTROLLER = 3;
        TYPE_ETHERNET        = 4;
        TYPE_TOKEN_RING      = 5;
        TYPE_SOUND           = 6;
        TYPE_PATA_CONTROLLER = 7;
        TYPE_SATA_CONTROLLER = 8;
        TYPE_SAS_CONTROLLER  = 9;
    }

    message Item
    {
        string description = 1;
        Type type          = 2;
        bool enabled       = 3;
    }

    repeated Item item = 1;
}
